### PR TITLE
OpenShift CLI start-build command

### PR DESCRIPTION
**Pull request by asafgabai**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: artifactory/commands/oc/startbuild.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+	trimmedVersion := strings.TrimPrefix(ocVersionStr, "v")
+	ocVersion := version.NewVersion(trimmedVersion)
+	if ocVersion.Compare(minSupportedOcVersion) > 0 {
+		return errorutils.CheckError(errors.New(fmt.Sprintf(
+			"JFrog CLI oc-start-build command requires OpenShift CLI version " + minSupportedOcVersion + " or higher")))
+	}
+	return nil
+}
+
+func extractOcOptionsFromArgs(args []string) (repo, serverId string, threads int, buildConfig *utils.BuildConfiguration, cleanArgs []string, err error) {
```
Comment (eyalbe4) [inline artifactory/commands/oc/startbuild.go:154]: It makes more sense for the flags extraction, including the validation to be done in jfrog-cli and not in jfrog-cli-core.

File: artifactory/utils/container/buildinfo.go [hunk 2]
```diff
@@ -338,15 +342,19 @@ func searchManifestAndLayersDetails(builder *buildInfoBuilder, imagePathPattern
...
 		} else {
-			manifestContent, err = verifyManifestByDigest(*searchesult, builder)
+			manifestContent, err = verifyManifestByDigest(*searchResult, builder)
 		}
 	} else {
+		if builder.containerManager == nil {
+			err = errorutils.CheckError(errors.New("build info collection for multi-architecture images is not supported in build-docker-create and oc-start-build commands"))
+			return
+		}
 		// Check if search results contain fat-manifest.
```
Comment (eyalbe4) [inline artifactory/utils/container/buildinfo.go:358]: // Check if search results contain fat-manifest.
-->
// Check if search results contain multi-architecture images (fat-manifest).

File: artifactory/utils/oc/getimagedetails.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/pkg/errors"
+	"strings"
+
+	gofrogcmd "github.com/jfrog/gofrog/io"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+func GetImageDetails(executablePath, ocBuildName string) (imageTag, manifestSha256 string, err error) {
+	getImageDetailsCmdConfig := createGetImageDetailsCmdConfig(executablePath, ocBuildName)
+	output, err := gofrogcmd.RunCmdOutput(getImageDetailsCmdConfig)
```
Comment (eyalbe4) [inline artifactory/utils/oc/getimagedetails.go:14]: I recommend not to use gofrogcmd.RunCmdOutput whenever possible, due to the overhead it brings, but use go's command execution API directly. See how we executed jfrog-cli plugins as a reference.
The same goes for the code in oc/version.go.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: artifactory/commands/oc/startbuild.go (hunk 1 above)
Comment (asafgabai) [reply artifactory/commands/oc/startbuild.go:153]: I moved the extraction part to jfrog-cli, but I didn't move the validation part.
It validates the args passed to oc only, and not all of the args of the command (like repo, build-name, etc).

**Review by eyalbe4 — APPROVED** [review]

**Pull request merged**